rust/osm2lanes/src/tag/mod.rs

```diff
@@ -90,6 +92,26 @@ impl Tags {
         self.map.get(k.as_ref()).map(String::as_str)
     }
 
+    /// Get the value for the given key and parse it into T.
+    /// Add a `TagsToLanesMsg::Unsupported` if parsing fails.
+    pub fn get_parsed<K: AsRef<str>, T: FromStr>(
+        &self,
+        key: &K,
+        warnings: &mut RoadWarnings,
+    ) -> Option<T> {
```

Comment:
`Result<Option<T>, String>`

where the Err is the value that cannot be parsed?

or

`Result<Option<T>, E>`

where E comes from the underlying FromStr

Reply:
maybe Option<Result<T>> makes more sense?

Reply:
It would be `Result<T, E>` (which would be converted to an `Option` with `ok()` most of the time, because the error is already added to the `warnings`).